pkg/worker/lifecycle.go

```diff
@@ -211,6 +222,28 @@ func (s *Worker) RunContainer(request *types.ContainerRequest) error {
 	return nil
 }
 
+func (s *Worker) buildOrPullImage(request *types.ContainerRequest, containerId string, outputChan chan common.OutputMsg) error {
+	switch {
+	case request.Dockerfile != nil:
+		log.Printf("<%s> - lazy-pull failed, building image from Dockerfile\n", containerId)
+		buildCtxPath, err := getBuildContext(request)
+		if err != nil {
+			return err
+		}
+		if err := s.imageClient.BuildAndArchiveImage(context.TODO(), outputChan, *request.Dockerfile, request.ImageId, buildCtxPath); err != nil {
```

Comment:
I don't think request should have dockerfile on it directly. Shouldn't it be request.buildoptions.dockerfile?

Reply:
It can be. I embedded the build options struct so that its members could be directly accessed. We can name it so that the access path is as you describe though.

Reply:
Named it